src/network/networkd-link.c

```diff
@@ -124,6 +125,9 @@ bool link_ipv4ll_enabled(Link *link) {
         if (STRPTR_IN_SET(link->kind, "vrf", "wireguard", "ipip", "gre", "ip6gre", "ip6tnl", "sit", "vti", "vti6"))
                 return false;
 
+        if (IN_SET(link_get_ipvlan_mode(link), NETDEV_IPVLAN_MODE_L3, NETDEV_IPVLAN_MODE_L3S))
```

Comment:
> disable IPv4LL for ipvlan with L3 or L3S mode

Can you please add a comment why this change (the whole commit) is being done?

Reply:
Not sure, but IPv4LL fails with `-ENOBUFS` in `sendto()`. @ssahani Do you have any idea?

Reply:
Could you enable the kernel with ```CONFIG_IPVLAN_L3S``` and test. It seems to me that it may be disabled.

Reply:
I noticed that L3 and L3S mode does not support ARP. And IPv4LL is implemented with arp-util.c (though I have not followed what it does).

Reply:
So... if the reason is that it fails in tests, than please say that in the commit message.

> Could you enable the kernel with CONFIG_IPVLAN_L3S and test.

CONFIG_IPVLAN_L3S is automatic, if CONFIG_NETFILTER and CONFIG_IPVLAN are enabled.

Reply:
IPv4LL requires ARP, but L3 and L3S do not support them, so `sendto()` in `arp_send_packet()` fails. This is not only for tests, but it is required to ipvlan or ipvtap interfaces become `configured` operstate (though IPv4LL is disabled by default, so it may be a mis-configuration by a user). But, anyway, we should kindly disable IPv4LL on such mode.